### PR TITLE
[tests-only][full-ci]extending test cases for copying file when shared with secure viewer

### PR DESCRIPTION
**Pull request by S-Panta**
## Description
Updated test cases for sharing via secure role 
```feature
Scenario Outline: copying a file to a folder with no permissions
Scenario Outline: copying a file into a shared folder shared via viewer and secure viewer permissions as the share
Scenario Outline: copying a file out of a shared folder shared via secure viewer permissions as the share
 Scenario Outline: copying a folder between shares received from different users when one share is shared via Secure viewer
 Scenario Outline: copying a file between shares received from different users when one share is shared via Secure viewer

```

## Related Issue

- Part of https://github.com/owncloud/ocis/issues/9334

## How Has This Been Tested?

- test environment:
- locally and CI

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [x] Tests only (no source changes)

## Checklist:

- [ ] Code changes
- [ ] Unit tests added
- [x] Acceptance tests added
- [ ] Documentation ticket raised: <link>

### CODE REVIEW

File: tests/acceptance/features/coreApiWebdavProperties/copyFile.feature [hunk 1]
```diff
@@ -735,6 +757,26 @@ Feature: copy file
...
+    And user "Brian" has created folder "/testshare"
+    And user "Brian" has uploaded file with content "hello world" to "testshare/fileInsideShare.txt"
+    And user "Brian" has sent the following resource share invitation:
+      | resource        | testshare     |
+      | space           | Personal      |
+      | sharee          | Alice         |
+      | shareType       | user          |
+      | permissionsRole | Secure viewer |
+    When user "Alice" copies file "/Shares/testshare/fileInsideShare.txt" to "/fileInsideShare.txt" using the WebDAV API
+    Then the HTTP status code should be "500"
```
Comment (SagarGi) [inline tests/acceptance/features/coreApiWebdavProperties/copyFile.feature:773]: i don't think this is the expected status code `500` ?

File: tests/acceptance/features/coreApiWebdavProperties/copyFile.feature [hunk 2]
```diff
@@ -735,6 +757,26 @@ Feature: copy file
       | old              |
       | new              |
 
+
+  Scenario Outline: copying a file out of a shared folder shared via secure viewer permissions as the sharee
```
Comment (SagarGi) [inline tests/acceptance/features/coreApiWebdavProperties/copyFile.feature:761]:
```suggestion
  Scenario Outline: sharee copy a file from a shared folder, shared with secure viewer permission 
```

File: tests/acceptance/features/coreApiWebdavProperties/copyFile.feature [hunk 3]
```diff
@@ -692,6 +694,26 @@ Feature: copy file
       | old              |
       | new              |
 
+
+  Scenario Outline: copying a file into a shared folder shared via viewer and secure viewer permissions as the sharee
```
Comment (SagarGi) [inline tests/acceptance/features/coreApiWebdavProperties/copyFile.feature:698]:
```suggestion
  Scenario Outline: sharee copy a file into a shared folder, shared with viewer and secure viewer permissions
```

File: tests/acceptance/features/coreApiWebdavProperties/copyFile.feature [hunk 4]
```diff
@@ -815,6 +857,41 @@ Feature: copy file
...
+      | resource        | testshare1           |
+      | space           | Personal             |
+      | sharee          | Alice                |
+      | shareType       | user                 |
+      | permissionsRole | <permissions-role-2> |
+    When user "Alice" copies file "/Shares/testshare0/testshare0.txt" to "/Shares/testshare1/testshare0.txt" using the WebDAV API
+    Then the HTTP status code should be "<status-code>"
+    Examples:
+      | dav-path-version | permissions-role-1 | permissions-role-2 | status-code |
+      | old              | Secure viewer      | Editor             | 500         |
```
Comment (SagarGi) [inline tests/acceptance/features/coreApiWebdavProperties/copyFile.feature:884]: Also here status is `500` ?

File: tests/acceptance/features/coreApiWebdavProperties/copyFile.feature [hunk 5]
```diff
@@ -815,6 +857,41 @@ Feature: copy file
       | old              |
       | new              |
 
+
+  Scenario Outline: copying a file between shares received from different users when one share is shared via Viewer and Secure viewer permission
```
Comment (anon-pradip) [inline tests/acceptance/features/coreApiWebdavProperties/copyFile.feature:861]:
```suggestion
  Scenario Outline: copying a file between shares received from different users when one share is shared via Viewer and Secure viewer permission
```

In just above scenario, it is written as `secure viewer` and here in this scenario it is `Secure viewer`. Just casing things. :grin:  Can we please make those same in all scenarios ensuring consistency?

File: tests/acceptance/features/coreApiWebdavProperties/copyFile.feature (hunk 1 above)
Comment (S-Panta) [reply tests/acceptance/features/coreApiWebdavProperties/copyFile.feature:773]: This table summarizes my finding
https://github.com/owncloud/ocis/issues/9334#issuecomment-2182573751

Comment (SagarGi) [reply tests/acceptance/features/coreApiWebdavProperties/copyFile.feature:773]: I see according to the issue comment it should be 403. So better but it 403 and put it in blocked state until it get fixed.

File: tests/acceptance/features/coreApiWebdavProperties/copyFile.feature (hunk 2 above)
[the remark this answers is not in the document]
Comment (S-Panta) [reply tests/acceptance/features/coreApiWebdavProperties/copyFile.feature:761]: The language is similar to line 717(previous scenario)

**Comment by S-Panta** [comment]
Copying file from Secure view share to personal space returns 500 status code.
Related issue for that: https://github.com/owncloud/ocis/issues/9482
Therefore the PR is moved to blocked

File: tests/acceptance/features/coreApiWebdavProperties/copyFile.feature [hunk 6]
```diff
@@ -643,6 +645,27 @@ Feature: copy file
       | old              |
       | new              |
 
+
+  Scenario Outline: sharee copy a file into a shared folder, shared with viewer and secure viewer permissions
+    Given using <dav-path-version> DAV path
+    And user "Brian" has been created with default attributes and without skeleton files
```
Comment (saw-jan) [inline tests/acceptance/features/coreApiWebdavProperties/copyFile.feature:651]: isn't it same as above scenario? `Scenario Outline: copying a file to a folder with no permissions`

File: tests/acceptance/features/coreApiWebdavProperties/copyFile.feature [hunk 7]
```diff
@@ -688,6 +711,27 @@ Feature: copy file
       | old              |
       | new              |
 
+
+  Scenario Outline: sharee copy a file from a shared folder, shared with secure viewer permission
```
Comment (saw-jan) [inline tests/acceptance/features/coreApiWebdavProperties/copyFile.feature:715]:
```suggestion
  Scenario Outline: sharee copies a file from a shared folder, shared with secure viewer permission
```

File: tests/acceptance/features/coreApiWebdavProperties/copyFile.feature [hunk 8]
```diff
@@ -688,6 +711,27 @@ Feature: copy file
...
+    Given using <dav-path-version> DAV path
+    And user "Brian" has been created with default attributes and without skeleton files
+    And user "Brian" has created folder "/testshare"
+    And user "Brian" has uploaded file with content "hello world" to "testshare/fileInsideShare.txt"
+    And user "Brian" has sent the following resource share invitation:
+      | resource        | testshare     |
+      | space           | Personal      |
+      | sharee          | Alice         |
+      | shareType       | user          |
+      | permissionsRole | Secure viewer |
```
Comment (saw-jan) [inline tests/acceptance/features/coreApiWebdavProperties/copyFile.feature:725]: could add coverage for viewer role as well.

File: tests/acceptance/features/coreApiWebdavProperties/copyFile.feature [hunk 9]
```diff
@@ -801,6 +882,43 @@ Feature: copy file
...
+    And user "Carol" has sent the following resource share invitation:
+      | resource        | testshare1           |
+      | space           | Personal             |
+      | sharee          | Alice                |
+      | shareType       | user                 |
+      | permissionsRole | <permissions-role-2> |
+    And user "Alice" has a share "testshare1" synced
+    When user "Alice" copies file "/Shares/testshare0/folder_to_copy/" to "/Shares/testshare1/folder_to_copy/" using the WebDAV API
+    Then the HTTP status code should be "<status-code>"
+    Examples:
```
Comment (saw-jan) [inline tests/acceptance/features/coreApiWebdavProperties/copyFile.feature:909]: check the file existence

File: tests/acceptance/features/coreApiWebdavProperties/copyFile.feature [hunk 10]
```diff
@@ -801,6 +882,43 @@ Feature: copy file
       | old              |
       | new              |
 
+
+  Scenario Outline: copying a folder between shares received from different users when one share is shared via Secure viewer
```
Comment (saw-jan) [inline tests/acceptance/features/coreApiWebdavProperties/copyFile.feature:886]: Is this scenario similar to above new one? we can merge both

File: tests/acceptance/features/coreApiWebdavProperties/copyFile.feature [hunk 11]
```diff
@@ -814,6 +858,47 @@ Feature: copy file
...
+    And user "Alice" has a share "testshare1" synced
+    When user "Alice" copies folder "/Shares/testshare0/folder_to_copy/" to "/Shares/testshare1/folder_to_copy/" using the WebDAV API
+    Then the HTTP status code should be "<status-code>"
+    And as "Alice" folder "/Shares/testshare1/folder_to_copy/" should not exist
+    When user "Alice" copies file "/Shares/testshare0/testshare0.txt" to "/Shares/testshare1/testshare0.txt" using the WebDAV API
+    Then the HTTP status code should be "<status-code>"
+    And as "Alice" file "/Shares/testshare1/testshare0.txt" should not exist
+
+    Examples:
+      | dav-path-version | permissions-role-1 | permissions-role-2 | status-code |
```
Comment (saw-jan) [inline tests/acceptance/features/coreApiWebdavProperties/copyFile.feature:892]:
```suggestion
    Then the HTTP status code should be "403"
    And as "Alice" folder "/Shares/testshare1/folder_to_copy/" should not exist
    When user "Alice" copies file "/Shares/testshare0/testshare0.txt" to "/Shares/testshare1/testshare0.txt" using the WebDAV API
    Then the HTTP status code should be "403"
    And as "Alice" file "/Shares/testshare1/testshare0.txt" should not exist
    Examples:
      | dav-path-version | permissions-role-1 | permissions-role-2 |
```

**Review by saw-jan — APPROVED** [review]

**Review by phil-davis — APPROVED** [review]

**Pull request merged**